chain/events/events_called.go

```diff
@@ -146,21 +147,25 @@ func (e *calledEvents) checkNewCalls(ts *types.TipSet) {
 
 func (e *calledEvents) queueForConfidence(triggerId uint64, msg *types.Message, ts *types.TipSet) {
 	trigger := e.triggers[triggerId]
-	triggerH := ts.Height() + uint64(trigger.confidence)
+
+	// messages are not applied in the tipset they are included in
```

Comment:
We *could* use the `MessagesForTipSet` on the parent tipset to source messages for this, which would avoid us having to deal with the off-by-one bits here

Reply:
General comment, doesnt need to be addressed in this PR, but we need to audit how this code handles incoming messages, I want to make sure its using the deduplicated messages from the parent tipset, and not the individual block messages as they come in

Reply:
It handles the deduplication. Added a test just in case.